test/bin/explicit-repo.t/run.t

```diff
@@ -51,7 +51,7 @@ Therefore the list of repositories in the lockfile should have the a git repo
 that's the same as the previous git-repository path but end with "#$SHORT_HASH"
 
   $ opam show --just-file --raw -fx-opam-monorepo-opam-repositories ./explicit-repo.opam.locked > locked-repos
-  $ grep -Po ".+(?=$SHORT_HASH1)" locked-repos
+  $ perl -ne "print \$1 if /(.+)$SHORT_HASH1/" locked-repos
```

Comment:
If we need `perl` here, I would suggest writing our own clean-up processor in OCaml instead of depending on a fairly large interpreter.

Reply:
I've changed the tests back to use a combination of sed and grep rather than perl, but using the common features available in the gnu and bsd versions so they work on linux and macos.